// ngx-cookie/esm2015/lib/cookie-writer.service.js
import { DOCUMENT } from '@angular/common';
import { Inject, Injectable } from '@angular/core';
import { buildCookieString } from './utils';
import * as ɵngcc0 from '@angular/core';
export class CookieWriterService {
    constructor(document) {
        this.document = document;
    }
    readAllAsString() {
        return this.document.cookie || '';
    }
    write(name, value, options) {
        this.document.cookie = buildCookieString(name, value, options);
    }
}
CookieWriterService.ɵfac = function CookieWriterService_Factory(t) { return new (t || CookieWriterService)(ɵngcc0.ɵɵinject(DOCUMENT)); };
CookieWriterService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: CookieWriterService, factory: CookieWriterService.ɵfac });
CookieWriterService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CookieWriterService, [{
        type: Injectable
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }]; }, null); })();
